src/io/flutter/run/DaemonManager.java

```diff
@@ -0,0 +1,348 @@
+      }
+      else if (text.startsWith(STARTING_DAEMON)) {
+        enableDevicePolling(event.getProcessHandler());
+      }
+    }
+  }
+
+  public boolean isConnectionReady(FlutterDaemonRunState state) {
+    // TODO Identify which app from state.
+    return !debugPorts.isEmpty();
+  }
+
+  public int getPort(FlutterDaemonRunState state) {
+    // TODO Identify which app from state.
+    return debugPorts.values().iterator().next();
+  }
+
+  private void enableDevicePolling(@NotNull ProcessHandler handler) {
+    sendCommand(handler, "device.enable", null);
+  }
+
+  private void eventLogMessage(LogMessage log) {
+  }
+
+  private void eventLogMessage(AppLog log) {
+  }
+
+  private void eventDeviceAdded(DeviceAdded added) {
+    myDevices.put(added.id, added);
+    if (myDevices.size() == 1) {
+      AppStart params = new AppStart(added.id, ".");
+      sendCommand(myHandler, "app.start", params);
+    }
+  }
+
+  private void eventDeviceRemoved(DeviceRemoved removed) {
+    myDevices.remove(removed.id);
+  }
+
+  private void eventAppStarted(AppStarted started) {
+    myApps.put(started.appId, started);
+  }
+
+  private void eventAppStopped(AppStopped stopped) {
+    myApps.remove(stopped.appId);
+  }
+
+  private void eventDebugPort(AppDebugPort debugPort) {
+    debugPorts.put(debugPort.appId, debugPort.port);
+  }
+
+  private void sendCommand(@NotNull ProcessHandler handler, String command, Params params) {
+    OutputStream input = handler.getProcessInput();
+    if (input == null) {
+      LOG.error("No process input");
+      return;
+    }
+    int cmdId = myCommandId++;
+    Method method = new Method(command, params, cmdId);
+    try (FlutterStream str = new FlutterStream(input)) {
+      str.print("[");
+      str.print(GSON.toJson(method));
+      str.println("]");
+      myCommands.put(cmdId, method);
+    }
+  }
+
+  private void processEventText(final String text, @NotNull ProcessHandler handler) {
+    myCurrentJsonText = text;
+    try {
+      JsonParser jp = new JsonParser();
+      JsonElement elem = jp.parse(text);
+      dispatch(elem.getAsJsonObject(), handler);
+    }
+    catch (JsonSyntaxException ex) {
+      LOG.error(ex);
+    }
+  }
+
+  private void dispatch(JsonObject obj, @NotNull ProcessHandler handler) {
+    this.myHandler = handler;
+    JsonPrimitive primId = obj.getAsJsonPrimitive("id");
+    if (primId == null) {
+      // event
+      JsonPrimitive primEvent = obj.getAsJsonPrimitive("event");
+      if (primEvent == null) {
+        LOG.error("Invalid JSON from flutter: " + myCurrentJsonText);
+        return;
+      }
+      String eventName = primEvent.getAsString();
+      JsonObject params = obj.getAsJsonObject("params");
+      if (eventName == null || params == null) {
+        LOG.error("Bad event from flutter: " + myCurrentJsonText);
+        return;
+      }
+      Event eventHandler = eventHandler(eventName);
+      if (eventHandler == null) return;
```

Comment:
I ran everything through the formatter. The single-line returns are according to style. In this sense, it is like Dart.